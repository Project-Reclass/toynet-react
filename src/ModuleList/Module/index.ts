import Module from './Module';
export * from './Module';
export default Module;