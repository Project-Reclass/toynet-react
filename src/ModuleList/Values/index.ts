import Values from './Values';
export default Values;