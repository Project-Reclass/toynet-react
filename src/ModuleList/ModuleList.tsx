import React from 'react';
import Module, { IModule } from './Module/Module';

interface Data extends IModule {
  subModules: IModule[];
}

const ModuleList = ({ moduleData }: {moduleData: Data[]}) => {
  return (
    <div>
      {moduleData.map(data => (
        <Module {...data} />
      ))}
    </div>
  );
};

export default ModuleList;