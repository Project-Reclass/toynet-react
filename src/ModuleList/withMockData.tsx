/* eslint-disable no-magic-numbers */
import React from 'react';
import { ModuleTypes, IModule } from './Module/Module';

const mockData = [
  {
    id: 1,
    moduleId: 0,
    title: 'Sandbox Tutorial',
    progress: 80,
    type: ModuleTypes.PARENT,
    subModules: [
      {
        id: 1,
        moduleId: 1,
        title: 'Emulator',
        progress: 40,
        type: ModuleTypes.ARTICLE,
      },
      {
        id: 1,
        moduleId: 1,
        title: 'Emulator Practice',
        progress: 10,
        type: ModuleTypes.EMULATOR,
      },
      {
        id: 1,
        moduleId: 1,
        title: 'Quiz',
        progress: 100,
        type: ModuleTypes.QUIZ,
      },
    ],
  },
  {
    id: 2,
    moduleId: 0,
    title: 'Switches and Routers',
    progress: 0,
    type: ModuleTypes.PARENT,
    subModules: [
      {
        id: 1,
        moduleId: 2,
        title: 'Switch',
        progress: 0,
        type: ModuleTypes.ARTICLE,
      },
      {
        id: 1,
        moduleId: 2,
        title: 'Quiz',
        progress: 0,
        type: ModuleTypes.QUIZ,
      },
      {
        id: 1,
        moduleId: 2,
        title: 'Routers',
        progress: 0,
        type: ModuleTypes.ARTICLE,
      },
      {
        id: 1,
        moduleId: 2,
        title: 'Quiz',
        progress: 0,
        type: ModuleTypes.QUIZ,
      },
    ],
  },
];

interface Data extends IModule {
  subModules: IModule[];
}

export default (Component: React.ComponentType<{moduleData: Data[]}>) => () => {
  return <Component moduleData={mockData} />;
};