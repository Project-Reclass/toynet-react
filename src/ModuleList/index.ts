import withMockData from './withMockData';
import ModuleList from './ModuleList';

export default withMockData(ModuleList);