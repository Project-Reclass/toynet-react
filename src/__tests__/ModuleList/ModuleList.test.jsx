import React from 'react';

import renderer from 'react-test-renderer';
import ModuleList from 'src/ModuleList';

describe('The Module List', () => {
  it('should render and match snapshot', () => {
    const tree = renderer.create(<ModuleList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});