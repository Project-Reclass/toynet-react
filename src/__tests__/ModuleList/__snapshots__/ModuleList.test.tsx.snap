// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Module List should render and match snapshot 1`] = `
<div>
  <div
    class="css-1h7hrtx"
  >
    <div
      class="css-1xeo39u"
    >
      <div
        class="css-1l4w6pd"
      >
        <div
          class="css-qx149r"
        >
          <p
            class="css-l8myuy"
          >
            Welcome!
          </p>
          <p
            class="css-annqw3"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam scelerisque fermentum volutpat. Proin ut iaculis mauris. Aenean commodo mauris at justo faucibus, in vulputate arcu maximus. Maecenas bibendum nisl urna, nec pretium lacus dapibus et. Nullam a diam sollicitudin, dapibus ante nec, pulvinar diam. Pellentesque sit amet dignissim felis. Nulla interdum, lacus ac rhoncus posuere, tortor leo pharetra nulla, a pulvinar quam sapien sed mi. Fusce bibendum mauris massa, at molestie arcu venenatis at. Vivamus in tellus tempus, porttitor diam a, laoreet magna. Morbi pharetra justo risus, non auctor nisl interdum varius.
          </p>
        </div>
        <div
          class="css-j7qwjs"
        >
          <p
            class="css-bkdsaw"
          >
            Course Completion
          </p>
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="25"
            class="css-c15t6t"
            role="progressbar"
          >
            <svg
              class="css-24ggqe"
              viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
            >
              <circle
                class="css-9y0bww"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-track="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="0"
                stroke-width="11.111111111111112"
              />
              <circle
                class="css-h8bys5"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-indicator="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="235.61944901923448"
                stroke-width="11.111111111111112"
              />
            </svg>
            <div
              class="css-13dwcxr-CircularProgressLabel"
            >
              25%
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-gjrj21"
    >
      <div>
        <div
          class="css-1xeo39u"
        >
          <div
            class="css-11dqbsu"
          >
            <svg
              class="css-1axzmpk"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                fill="currentColor"
              />
            </svg>
            <p
              class="css-grke7g"
            >
              Example Module
            </p>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero css-vppuuz"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 250ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="css-0"
            >
              <div
                class="css-1opj1b2"
              >
                <div
                  class="css-k008qs"
                >
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-qt77o5"
                      type="check"
                    >
                      <svg
                        class="css-1sb13dz"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 14 14"
                      >
                        <g
                          fill="currentColor"
                        >
                          <polygon
                            points="5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039"
                          />
                        </g>
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-gmscv5"
                  >
                    <div
                      class="css-1wt6rk2"
                    >
                      <p
                        class="css-c02fy7"
                      >
                        <a
                          class="css-1xx1v7d"
                          href="/module/0/quiz/4001"
                        >
                          Quiz 1
                        </a>
                      </p>
                      <div
                        class="css-xtemij"
                      >
                        <p
                          class="css-5yrbsi"
                        >
                          Completed
                        </p>
                        <div
                          class="css-52gy4w"
                        >
                          <div
                            aria-valuemax="100"
                            aria-valuemin="0"
                            aria-valuenow="100"
                            class="css-1tacofv-Progress"
                            role="progressbar"
                          />
                        </div>
                      </div>
                    </div>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-1xc5jix"
                      type="lock"
                    >
                      <svg
                        class="css-1qqmyhf"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19.5,9.5h-.75V6.75A6.751,6.751,0,0,0,5.533,4.811a1.25,1.25,0,1,0,2.395.717A4.251,4.251,0,0,1,16.25,6.75V9a.5.5,0,0,1-.5.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-gmscv5"
                  >
                    <div
                      class="css-1wt6rk2"
                    >
                      <p
                        class="css-c02fy7"
                      >
                        <a
                          class="css-1xx1v7d"
                          href="/module/0/article/2001"
                        >
                          Ever wondered how underwater cables are laid?
                        </a>
                      </p>
                      <div
                        class="css-xtemij"
                      >
                        <p
                          class="css-5yrbsi"
                        >
                          Module progress
                        </p>
                        <div
                          class="css-52gy4w"
                        >
                          <div
                            aria-valuemax="100"
                            aria-valuemin="0"
                            aria-valuenow="10"
                            class="css-r1648o-Progress"
                            role="progressbar"
                          />
                        </div>
                      </div>
                    </div>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-1xc5jix"
                      type="lock"
                    >
                      <svg
                        class="css-1qqmyhf"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19.5,9.5h-.75V6.75A6.751,6.751,0,0,0,5.533,4.811a1.25,1.25,0,1,0,2.395.717A4.251,4.251,0,0,1,16.25,6.75V9a.5.5,0,0,1-.5.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-gmscv5"
                  >
                    <div
                      class="css-gg4vpm"
                    >
                      <p
                        class="css-c02fy7"
                      >
                        <a
                          class="css-1xx1v7d"
                          href="/value/5001"
                        >
                          Values - Integrity
                        </a>
                      </p>
                      <div
                        class="css-xtemij"
                      >
                        <p
                          class="css-5yrbsi"
                        >
                          Module progress
                        </p>
                        <div
                          class="css-52gy4w"
                        >
                          <div
                            aria-valuemax="100"
                            aria-valuemin="0"
                            aria-valuenow="0"
                            class="css-1u0whdh-Progress"
                            role="progressbar"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          class="css-1xeo39u"
        >
          <div
            class="css-11dqbsu"
          >
            <svg
              class="css-1axzmpk"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
                fill="currentColor"
              />
            </svg>
            <p
              class="css-grke7g"
            >
              Second Module
            </p>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="rah-static rah-static--height-zero css-vppuuz"
          style="height: 0px; overflow: hidden;"
        >
          <div
            style="transition: opacity 250ms ease 0ms; opacity: 0; display: none;"
          >
            <div
              class="css-0"
            >
              <div
                class="css-1opj1b2"
              >
                <div
                  class="css-k008qs"
                >
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-1xc5jix"
                      type="lock"
                    >
                      <svg
                        class="css-1qqmyhf"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19.5,9.5h-.75V6.75A6.751,6.751,0,0,0,5.533,4.811a1.25,1.25,0,1,0,2.395.717A4.251,4.251,0,0,1,16.25,6.75V9a.5.5,0,0,1-.5.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-gmscv5"
                  >
                    <div
                      class="css-1wt6rk2"
                    >
                      <p
                        class="css-c02fy7"
                      >
                        <a
                          class="css-1xx1v7d"
                          href="/module/0/quiz/4002"
                        >
                          Quiz 2
                        </a>
                      </p>
                      <div
                        class="css-xtemij"
                      >
                        <p
                          class="css-5yrbsi"
                        >
                          Module progress
                        </p>
                        <div
                          class="css-52gy4w"
                        >
                          <div
                            aria-valuemax="100"
                            aria-valuemin="0"
                            aria-valuenow="0"
                            class="css-1u0whdh-Progress"
                            role="progressbar"
                          />
                        </div>
                      </div>
                    </div>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-1xc5jix"
                      type="lock"
                    >
                      <svg
                        class="css-1qqmyhf"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19.5,9.5h-.75V6.75A6.751,6.751,0,0,0,5.533,4.811a1.25,1.25,0,1,0,2.395.717A4.251,4.251,0,0,1,16.25,6.75V9a.5.5,0,0,1-.5.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-gmscv5"
                  >
                    <div
                      class="css-1wt6rk2"
                    >
                      <p
                        class="css-c02fy7"
                      >
                        <a
                          class="css-1xx1v7d"
                          href="/module/0/article/2002"
                        >
                          Father of the internet, Vint Cerf, on creating the interplanetary internet
                        </a>
                      </p>
                      <div
                        class="css-xtemij"
                      >
                        <p
                          class="css-5yrbsi"
                        >
                          Module progress
                        </p>
                        <div
                          class="css-52gy4w"
                        >
                          <div
                            aria-valuemax="100"
                            aria-valuemin="0"
                            aria-valuenow="50"
                            class="css-1o3mzkk-Progress"
                            role="progressbar"
                          />
                        </div>
                      </div>
                    </div>
                    <hr
                      class="css-1fic0ux"
                    />
                  </div>
                </div>
                <div
                  class="css-k008qs"
                >
                  <div
                    class="css-j7qwjs"
                  >
                    <div
                      class="css-1xc5jix"
                      type="lock"
                    >
                      <svg
                        class="css-1qqmyhf"
                        focusable="false"
                        role="presentation"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19.5,9.5h-.75V6.75A6.751,6.751,0,0,0,5.533,4.811a1.25,1.25,0,1,0,2.395.717A4.251,4.251,0,0,1,16.25,6.75V9a.5.5,0,0,1-.5.5H4.5a2,2,0,0,0-2,2V22a2,2,0,0,0,2,2h15a2,2,0,0,0,2-2V11.5A2,2,0,0,0,19.5,9.5Zm-9.5,6a2,2,0,1,1,3,1.723V19.5a1,1,0,0,1-2,0V17.223A1.994,1.994,0,0,1,10,15.5Z"
                          fill="currentColor"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="css-gmscv5"
                  >
                    <div
                      class="css-gg4vpm"
                    >
                      <p
                        class="css-c02fy7"
                      >
                        <a
                          class="css-1xx1v7d"
                          href="/value/5002"
                        >
                          Values - Respect
                        </a>
                      </p>
                      <div
                        class="css-xtemij"
                      >
                        <p
                          class="css-5yrbsi"
                        >
                          Module progress
                        </p>
                        <div
                          class="css-52gy4w"
                        >
                          <div
                            aria-valuemax="100"
                            aria-valuemin="0"
                            aria-valuenow="0"
                            class="css-1u0whdh-Progress"
                            role="progressbar"
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
